lib/vrl/web-playground/public/index.html

```diff
@@ -66,180 +66,11 @@ <h1>VRL Playground</h1>
-        if (window.eventEditor.getModel().getLineCount() > 1) {
-          let lines = window.eventEditor.getModel().getLinesContent();
-          // if the second line is a json object
-          // we assume the user has passed in valid json on each
-          // line
-          if (lines[1][0] == "{" && lines[1][lines[1].length - 1] == "}") {
-            return true;
-          }
-        }
-        return false;
-      }
-      function handleRunCode(input) {
-        if (isJsonL()) {
-          let inputs = [];
-          let program = window.programEditor.getValue();
-          let lines = window.eventEditor.getModel().getLinesContent();
-          lines.forEach((line) => {
-            inputs.push({
-              program: program,
-              event: tryJsonParse(line)
-            })
-          });
-
-          let results = [];
-          inputs.forEach((input) => {
-            results.push(window.run_vrl(input));
-          })
-          let outputs = [];
-          results.forEach((res) => {
-            if (res.output) {
-              outputs.push(JSON.stringify(res["result"], null, "\t"));
-            } else if (res.msg) {
-              outputs.push(res["msg"]);
-            }
-          })
-          // disable output validation for json since jsonl input won't ouput valid json
-          monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
-            validate: false
-          });
-          window.outputEditor.setValue(outputs.join("\n"));
-          return results;
-        }
-
-        if (input == null) {
-          input = {
-            program: window.programEditor.getValue(),
-            event: tryJsonParse(window.eventEditor.getValue()),
-          };
-        }
-
-        let res = window.run_vrl(input);
-
-        console.log("[DEBUG::handleRunCode()] Printing out res: ", res);
-        if (res.output) {
-          window.outputEditor.setValue(JSON.stringify(res["result"], null, "\t"));
-        } else if (res.msg) {
-          // disable output validation for json
-          // since vrl error msgs won't ouput json
-          monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
-            validate: false
-          });
-          window.outputEditor.setValue(res["msg"]);
-        }
-        return res;
-      }
-    </script>
-    <script>
-      function handleShareCode() {
-        let state = {
-          program: window.programEditor.getValue(),
-          event: JSON.parse(window.eventEditor.getValue()),
-        };
-
-        console.log("[DEBUG::handleShareCode()] Printing out state", state);
-        console.log(
-          "[DEBUG::handleShareCode()] Printing out base64 encoded state\n",
-          btoa(JSON.stringify(state))
-        );
-        window.history.pushState(state, "", `?state=${encodeURIComponent(btoa(JSON.stringify(state)))}`);
-      }
+    <script type="module">
+      import * as vrl from "./index.js";
+      window.vrlPlayground = new vrl.VrlWebPlayground();
```

Comment:
Why put this in a separate `<script>` tag instead of just putting it at the end of `index.js`?

Reply:
Ah I think this was a lack of knowledge what the module tag means in javascript. I was under the assumption we couldn't use variables created from modules outside of javascript. But after adding the most recent commit seems like the playground still functions.